Add doc for `mapInArrow` [skip ci] (#6974)
NVIDIA/spark-rapids: docs/additional-functionality/rapids-udfs.md

To find details on the above Python configuration settings, please see the 
[RAPIDS Accelerator for Apache Spark Configuration Guide](../configs.md). 
Search 'pandas' for a quick navigation jump.

### mapInArrow
[mapInArrow](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.mapInArrow.html#pyspark.sql.DataFrame.mapInArrow) is a PySpark API introduced in Spark 3.3.0.

> And the RAPIDS Accelerator also supports acceleration for it in the same way with the Pandas UDF.

Reviewer: ```suggestion
The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way with the Pandas UDF.
```

> The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way with the Pandas UDF.

Reviewer: Sorry, one more minor suggestion: 
```suggestion
The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way as Pandas UDF.
```
